tools/toolchain/Dockerfile

```diff
@@ -14,5 +14,17 @@ RUN dnf -y update \
     && echo 'Defaults !requiretty' >> /etc/sudoers
 RUN dnf -y install llvm
 RUN mkdir -p /root/.m2/repository
+
+FROM base AS bolt
+RUN git clone https://github.com/llvm/llvm-project --branch llvmorg-14.0.3 --depth 1
+ADD ./tools/toolchain/bolt-fix.patch ./
+RUN env GIT_COMMITTER_NAME='ScyllaDB' GIT_COMMITTER_EMAIL='scylla@example.com' git -C llvm-project am ../bolt-fix.patch
+WORKDIR llvm-project
+RUN cmake -B build -S llvm -DLLVM_ENABLE_PROJECTS='bolt' -G Ninja -DLLVM_TARGETS_TO_BUILD=X86 -DCMAKE_BUILD_TYPE=Release -DCMAKE_C_COMPILER=/usr/bin/clang -DCMAKE_CXX_COMPILER=/usr/bin/clang++ -DLLVM_CCACHE_BUILD=yes -DLLVM_ENABLE_LLD=yes -DLLVM_BUILD_RUNTIME=No
+RUN cmake --build build --target install
+
+FROM base
+COPY --from=bolt /usr/local/bin/llvm-bolt /usr/local/bin/merge-fdata /usr/local/bin
+COPY --from=bolt /usr/local/lib/libbolt_rt_instr.a /usr/local/lib
```

Comment:
There's a problem here in that you bypass ./install-dependencies.sh. Dockerfile was meant as a way to package the dependencies, not a way to specify or generate them.

Maybe we should move it there (with a big rm -rf on the build directory to clean up).